src/createAnimatedComponent/JSPropsUpdater.ts

```diff
@@ -0,0 +1,170 @@
+'use strict';
+import {
+  NativeEventEmitter,
+  NativeModules,
+  findNodeHandle,
+} from 'react-native';
+import {
+  nativeShouldBeMock,
+  shouldBeUseWeb,
+} from '../reanimated2/PlatformChecker';
+import type { StyleProps } from '../reanimated2';
+import { runOnJS, runOnUIImmediately } from '../reanimated2/threads';
+import type {
+  AnimatedComponentProps,
+  IAnimatedComponentInternal,
+  IJSPropsUpdater,
+  InitialComponentProps,
+} from './commonTypes';
+
+interface ListenerData {
+  viewTag: number;
+  props: StyleProps;
+}
+
+class JSPropsUpdaterPaper implements IJSPropsUpdater {
+  private static _tagToComponentMapping = new Map();
+  private _reanimatedEventEmitter: NativeEventEmitter;
+  private static _reanimatedModuleMock = {
+    async addListener(): Promise<void> {
+      // noop
+    },
+    async removeListeners(): Promise<void> {
+      // noop
+    },
+  };
+
+  constructor() {
+    let reanimatedModule: typeof JSPropsUpdaterPaper._reanimatedModuleMock;
+    if (nativeShouldBeMock()) {
+      reanimatedModule = JSPropsUpdaterPaper._reanimatedModuleMock;
+    } else {
+      reanimatedModule = NativeModules.ReanimatedModule;
+    }
+    this._reanimatedEventEmitter = new NativeEventEmitter(reanimatedModule);
+  }
+
+  public addOnJSPropsChangeListener(
+    animatedComponent: React.Component<
+      AnimatedComponentProps<InitialComponentProps>
+    > &
+      IAnimatedComponentInternal
+  ) {
+    const viewTag = findNodeHandle(animatedComponent);
```

Comment:
Let's just make sure that we never use `findNodeHandle` on web.

Reply:
I think this is enough to keep it safe:
```js
let JSPropsUpdater: JSPropsUpdaterOptions;
if (SHOULD_BE_USE_WEB) {
  JSPropsUpdater = JSPropsUpdaterWeb;
} else if (global._IS_FABRIC) {
  JSPropsUpdater = JSPropsUpdaterFabric;
} else {
  JSPropsUpdater = JSPropsUpdaterPaper;
}
export default JSPropsUpdater;
```